game/analysis.py

```
@@ -1,52 +1,31 @@
+import os
+import json
+
+# Load the opening book JSON once on startup
+BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+OPENINGS_JSON_PATH = os.path.join(BASE_DIR, 'openings.json')
+
+try:
+    with open(OPENINGS_JSON_PATH, 'r', encoding='utf-8') as f:
```

Ruff (0.15.15)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)